app/services/search_orders.rb

```diff
@@ -24,7 +24,12 @@ def pagination_data
   attr_reader :params, :current_user
 
   def fetch_orders
-    @search = ::Permissions::Order.new(current_user).editable_orders.ransack(params[:q])
+    @search = ::Permissions::Order.new(current_user)
+                                  .editable_orders
+                                  .joins(shipments: { shipping_rates: :shipping_method })
+                                  .where(spree_shipping_rates: { selected: true })
+                                  .select('spree_orders.*, spree_shipping_rates.shipping_method_id as shipping_method_id')
+                                  .ransack(params[:q])
```

Comment:
Nice one @jeduardo824 almost there!

I am not sure how to do this, @Matt-Yorkley can you advise here?

one option here is to bypass ransack for this attribute. remove it from params[:q] and send it as a normal param and them use that param to add an extra filter to the query like:
```ruby
.joins(shipments: :shipping_rates })
.where("spree_shipping_rates.selected == true and spree_shipping_rates.shipping_method_id = #{params[shipping_method_id]}")
```
btw, we dont need to join shipping_method, shipping_method_id in shipping_rates is enough to filter

Reply:
I thought about doing that. My concern is when the user doesn't use the Shipping Method filter. I could create a scope to return based on the shipping_method_id and use that only when params[:shipping_method_id] is not null.

Reply:
yes, you can build the query incrementally, I wouldnt create a scope, I'd just do it here:
```
 def fetch_orders
    @search = ::Permissions::Order.new(current_user)
                                  .editable_orders
    if params[shipping_method_id]
         @search = @search.joins(shipments: :shipping_rates })
                            .where("spree_shipping_rates.selected == true and spree_shipping_rates.shipping_method_id = #{params[shipping_method_id]}")
     end

      @search = @search.ransack(params[:q])

```

Reply:
Good idea. I will try and update my PR.

Reply:
It worked!

Reply:
awesome! some errors in the build, can you have a look?

Reply:
Sure